Frontend/ui-customer/src/Components/DriverInfoCard/View.purs

```diff
@@ -246,26 +246,35 @@ estimatedTimeAndDistanceView push state =
 
 otpView :: forall w. (Action -> Effect Unit) -> DriverInfoCardState -> PrestoDOM ( Effect Unit) w
 otpView push state =
-  linearLayout
-  [ height WRAP_CONTENT
-  , width WRAP_CONTENT
-  ] (map(\item ->
-      linearLayout
-        [ height $ V 32
-        , width $ V 32
-        , gravity CENTER
-        , cornerRadius 4.0
-        , accessibility DISABLE
-        , background state.data.config.quoteListModel.otpTextBackground
-        , margin $ MarginLeft 7
-        ][ textView (
-            [ height WRAP_CONTENT
-            , width WRAP_CONTENT
-            , text item
-            , accessibility DISABLE
-            , color state.data.config.quoteListModel.otpTextColor
-            ] <> FontStyle.h2 TypoGraphy)
-        ]) $ split (Pattern "")  state.data.otp)
+  parentView os $ 
+  [ linearLayout
+    [ height WRAP_CONTENT
+    , width WRAP_CONTENT
+    , gravity CENTER
+    ] (map(\item ->
+        linearLayout
+          [ height $ V 32
+          , width $ V 32
+          , gravity CENTER
+          , cornerRadius 4.0
+          , accessibility DISABLE
+          , background state.data.config.quoteListModel.otpTextBackground
+          , margin $ MarginLeft 7
+          ][ textView (
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text item
+              , accessibility DISABLE
+              , color state.data.config.quoteListModel.otpTextColor
+              ] <> FontStyle.h2 TypoGraphy)
+          ]) $ split (Pattern "") state.data.otp)
+  ]
+  where 
+    parentView os = ( (if os == "IOS" then linearLayout else horizontalScrollView) $
+                        [ height WRAP_CONTENT
+                        , width WRAP_CONTENT
+                        , gravity CENTER
+                        ] <> if os == "IOS" then [] else [scrollBarX false] )
```

Comment:
Looks like no need of condition on this extra prop It can be added in body itself, In case of linearLayout It will not work 
` <> if os == "IOS" then [] else [scrollBarX false]`